src/ui/chart-list.jsx

```diff
@@ -0,0 +1,241 @@
+      const tooltip = (
+        <div className="tooltip">
+          <div className="tooltip-title">Chart Leads To</div>
+          <div>{mappedIslandForChart}</div>
+        </div>
+      );
+
+      chartContent = (
+        <Tooltip tooltipContent={tooltip}>
+          {chartElement}
+        </Tooltip>
+      );
+    } else {
+      chartContent = chartElement;
+    }
+
+    return <td key={chart}>{chartContent}</td>;
+  }
+
+  chart(chartName, showLocationTooltip = true) {
+    if (_.isNil(chartName)) {
+      return null;
+    }
+
+    const {
+      decrementItem,
+      incrementItem,
+      spheres,
+      trackerState,
+      trackSpheres,
+    } = this.props;
+
+    const itemCount = trackerState.getItemValue(chartName);
+
+    const color = itemCount === 1
+      ? LogicCalculation.LOCATION_COLORS.CHECKED_LOCATION
+      : LogicCalculation.LOCATION_COLORS.AVAILABLE_LOCATION;
+
+    let locations = [];
+    if (showLocationTooltip && trackSpheres) {
+      locations = trackerState.getLocationsForItem(chartName);
+    }
+
+    const incrementItemFunc = (event) => {
+      event.stopPropagation();
+
+      incrementItem(chartName);
+    };
+
+    const decrementItemFunc = (event) => {
+      event.preventDefault();
+
+      decrementItem(chartName);
+    };
+
+    const chartElement = (
+      <div
+        className={`detail-span ${color} font-smallest`}
+        onClick={incrementItemFunc}
+        onContextMenu={decrementItemFunc}
+        onKeyDown={KeyDownWrapper.onSpaceKey(incrementItemFunc)}
+        role="button"
+        tabIndex="0"
+      >
+        {chartName}
+      </div>
+    );
+
+    const foundAtTooltipContent = !_.isEmpty(locations)
+      ? <FoundAtTooltip locations={locations} spheres={spheres} />
+      : null;
+
+    const mappedIslandForChart = trackerState.getIslandForChart(chartName);
+    const chartLeadsTo = !_.isNil(mappedIslandForChart) ? (
+      <div className="tooltip">
+        <div className="tooltip-title">Chart Leads To</div>
+        <div>{mappedIslandForChart}</div>
+      </div>
+    ) : null;
+
+    let outerChartElement;
+    if (foundAtTooltipContent || chartLeadsTo) {
+      const tooltipContent = (
+        <>
+          {foundAtTooltipContent}
+          {chartLeadsTo}
+        </>
+      );
+      outerChartElement = (
+        <td key={chartName}>
+          <Tooltip
+            tooltipContent={tooltipContent}
+          >
+            {chartElement}
+          </Tooltip>
+        </td>
+      );
+    } else {
+      outerChartElement = <td key={chartName}>{chartElement}</td>;
+    }
+
+    return outerChartElement;
+  }
+
+  render() {
+    const { clearOpenedMenus, openedChartForIsland } = this.props;
+    const treasureCharts = _.sortBy(_.filter(CHARTS, (o) => o.includes('Treasure Chart')), (o) => LogicHelper.parseChartNumber(o));
+    const triforceCharts = _.sortBy(_.filter(CHARTS, (o) => o.includes('Triforce Chart')), (o) => LogicHelper.parseChartNumber(o));
```

Comment:
I think it would be better if these were two separate methods in `LogicHelper` - `allTreasureCharts` and `allTriforceCharts`.

An additional thing to consider - rather than filtering `CHARTS` and parsing the chart numbers, you could instead do a loop across the chart numbers (1 to 41 for treasure charts, 1 to 8 for triforce charts) and then generate the strings based on that. You could store `8` as a constant in `LogicHelper` (`NUM_TRIFORCE_CHARTS`) and then use `CHARTS.length - this.NUM_TRIFORCE_CHARTS` to get to `41`.